File: 20.ValidParanthesis.py
class Solution:
    def isValid(self, s: str) -> bool:
        chars = []
        i = 0
        while i != len(s):
            top_index = len(chars) - 1
            top = chars[top_index] if len(chars) > 0 else ' '
            ch = s[i]
            if ch == '(' or ch == '{' or ch == '[':
                chars.append(s[i])
            elif len(chars) == 0:
                return False
            elif ch == ')' and top == '(':
                chars.pop()
            elif ch == '}' and top == '{':
                chars.pop()
            elif ch == ']' and top == '[':
                chars.pop()
            else:
                chars.append(ch)
            i+=1
        print(chars)
        return True if len(chars) == 0 else False

File: 20.ValidParanthesis.java
class Solution {

    public boolean isValid(String s) {
        List<Character> chars = new ArrayList<>();
        int i = 0;
        while (i != s.length()) {
            char ch = s.charAt(i);
            int top_index = chars.size() - 1;
            char top = chars.size() > 0 ? chars.get(top_index) : ' ';
             if (ch == '(' || ch == '{' || ch == '[') {
                chars.add(ch);
             } else if(chars.size() == 0) {
                 return false;
             } else if (ch == ')' && top == '(') {
                 char removedChar = chars.remove(top_index);
             } else if(ch == '}' && top == '{') {
                 char removedChar = chars.remove(top_index);
             } else if (ch == ']' && top == '[') {
                 char removedChar = chars.remove(top_index);
             } else {
                 chars.add(ch);
             }
             i++;
        }

        return chars.isEmpty();
        
        
    }
}
